src/GitTools.jl

```diff
@@ -109,7 +109,11 @@ function clone(io::IO, url, source_path; header=nothing, credentials=nothing, kw
     end
     try
         if use_cli_git()
-            run(pipeline(`git clone --quiet $url $source_path`; stdout=devnull))
+            try
+                run(pipeline(`git clone --quiet $url $source_path`; stdout=devnull))
+            catch err
+                Pkg.Types.pkgerror("Git cli failed to clone from '$(url)', error: $err")
```

Comment:
What about showing the command?
```suggestion
                Pkg.Types.pkgerror("`git clone $url` failed, error: $err")
```